src/libtiled/mapreader.cpp

```diff
@@ -716,8 +723,26 @@ void MapReaderPrivate::readTilesetWangSets(Tileset &tileset)
             const QString name = atts.value(QLatin1String("name")).toString();
             const WangSet::Type type = wangSetTypeFromString(atts.value(QLatin1String("type")).toString());
             const int tile = atts.value(QLatin1String("tile")).toInt();
+            const bool randomizeOrientation = atts.value(QLatin1String("randomizeOrientation")).toInt()!=0;
 
             auto wangSet = std::make_unique<WangSet>(&tileset, name, type, tile);
+            if (canrotate)
+            {
+                wangSet->setAsNeededFlipHorizontally(true);
+                wangSet->setAsNeededFlipVertically(true);
+                wangSet->setAsNeededFlipAntiDiagonally(true);
+            }
+            else
+            {
+                const bool flipHorizontally = atts.value(QLatin1String("flipHorizontally")).toInt()!=0;
+                const bool flipVertically = atts.value(QLatin1String("flipVertically")).toInt()!=0;
+                const bool flipAntiDiagonally = atts.value(QLatin1String("flipAntiDiagonally")).toInt()!=0;
+                wangSet->setAsNeededFlipHorizontally(flipHorizontally);
+                wangSet->setAsNeededFlipVertically(flipVertically);
+                wangSet->setAsNeededFlipAntiDiagonally(flipAntiDiagonally);
+            }
+            if (alternaterotation || randomizeOrientation)
+                wangSet->setRandomizeOrientation(true);
```

Comment:
I think it might be more intuitive to negate this and call it "prefer non-transformed tiles".

Reply:
Good point, after looking at these for some days I was less and less able to come up with a good names.